src/main/scala/nodes/learning/KernelGenerator.scala

```diff
@@ -0,0 +1,163 @@
+package nodes.learning
+
+import scala.reflect.ClassTag
+
+import breeze.linalg._
+import breeze.math._
+import breeze.numerics._
+
+import org.apache.spark.rdd.RDD
+
+import pipelines.Logging
+import workflow.Transformer
+import utils._
+
+/**
+ * Base trait for functions that compute a kernel on a given dataset.
+ */
+trait KernelGenerator[T] {
+
+  /**
+   * Create a kernel transformer using the given dataset as one of the arguments
+   * to the kernel function. That is, if the kernel function is \phi(x, y), this
+   * binds one of the arguments using the given data.
+   *
+   * @param trainData training data matrix to be used for kernel transformation
+   * @return a kernel transformer that can be applied to train or test data
+   */
+  def fit(trainData: RDD[T]): KernelTransformer[T]
```

Comment:
If we made `KernelTransformer extends Transformer[T,DenseVector[Double]]`, and have a `KernelTransformer` also need this special `apply(data: RDD[T]): KernelMatrix` interface (maybe we call it `computeKernelMatrix(data: RDD[T]): KernelMatrix`), then maybe we could build up kernel pipelines with `Features andThen KernelGenerator andThen KernelSolver` as opposed to what we're doing below? 

I'm just trying to fit the kernel stuff into our current APIs rather than inventing new ones if we can help it. I also wonder if we should  be introducing a `KernelSolver` that generally knows how to take `KernelMatrix` as input and emit a `KernelModel` which is a thing that is parameterized by the `KernelTransformer` it uses.